srtcore/api.cpp

```diff
@@ -2090,6 +2121,135 @@ CUDT::APIError::APIError(CodeMajor mj, CodeMinor mn, int syserr)
     CUDT::s_UDTUnited.setError(new CUDTException(mj, mn, syserr));
 }
 
+// This is an internal function; 'type' should be pre-checked if it has a correct value.
+// This doesn't have argument of GroupType due to header file conflicts.
+CUDTGroup& CUDT::newGroup(int type)
+{
+    CGuard guard (s_UDTUnited.m_IDLock);
+    SRTSOCKET id = s_UDTUnited.generateSocketID(true);
+
+    // Now map the group
+    return s_UDTUnited.addGroup(id).id(id).type(SRT_GROUP_TYPE(type));
+}
+
+SRTSOCKET CUDT::createGroup(SRT_GROUP_TYPE gt)
+{
+    // Doing the same lazy-startup as with srt_create_socket()
+    if (!s_UDTUnited.m_bGCStatus)
+        s_UDTUnited.startup();
+
+    try
+    {
+        return newGroup(gt).id();
+    }
+    catch (CUDTException& e)
+    {
+        return APIError(e);
+    }
+    catch (std::bad_alloc& e)
+    {
+        return APIError(MJ_SYSTEMRES, MN_MEMORY, 0);
+    }
+}
```

Comment:
The function misses the normal return path. Although it should never happen, some static analyzers might complain.
I don't see a nice way to overcome this though. Don't want to add an unneccessary variable.

Reply:
Easy. Just add a dumb `return SRT_INVALID_SOCK` that will never happen.

Reply:
Argh!!! So easy 😆